src-tauri/src/lib.rs

```diff
@@ -314,6 +319,97 @@ pub fn launch_northstar(
     ))
 }
 
+pub fn get_titanfall_proton() -> Option<String> {
+    let steampath = SteamDir::locate().unwrap();
+    let configpath = steampath.path.join("config").join("config.vdf");
+
+    let vdf_text = fs::read_to_string(configpath).unwrap();
+    let config = Vdf::parse(&vdf_text).unwrap().value;
+    let obj = config.unwrap_obj();
+
+    let software = obj.get("Software")?.get(0)?.get_obj()?;
+    let valve = software.get("Valve")?.get(0)?.get_obj()?;
+    let steam = valve.get("Steam")?.get(0)?.get_obj()?;
+    let compat_tool_mapping = steam.get("CompatToolMapping")?.get(0)?.get_obj()?;
+    let tfapp = compat_tool_mapping
+        .get(TITANFALL2_STEAM_ID)?
+        .get(0)?
+        .get_obj()?;
+    let name = tfapp.get("name")?.get(0)?.get_str()?.to_string();
+
+    return Some(name);
+}
+
+pub fn launch_northstar_steam(
+    game_install: GameInstall,
+    _bypass_checks: Option<bool>,
+) -> Result<String, String> {
+    if !matches!(game_install.install_type, InstallType::STEAM) {
+        return Err("Titanfall2 was not installed via Steam".to_string());
+    }
+
+    if get_host_os() != "windows" {
+        match get_titanfall_proton() {
+            Some(proton) => {
+                if !proton.to_ascii_lowercase().contains("northstarproton") {
+                    return Err("Titanfall2 was not configured to use NorthstarProton".to_string());
+                }
+            }
+            None => {
+                return Err("Titanfall2 was not configured to use a compatibility tool".to_string());
+            }
+        }
+    }
+
+    // Switch to Titanfall2 directory to set everything up
+    if std::env::set_current_dir(game_install.game_path.clone()).is_err() {
+        // We failed to get to Titanfall2 directory
+        return Err("Couldn't access Titanfall2 directory".to_string());
+    }
+
+    let run_northstar = "run_northstar.txt";
+    let run_northstar_bak = "run_northstar.txt.bak";
+
+    if Path::new(run_northstar).exists() {
+        // rename should ovewrite existing files
+        // TODO: deal with folders
```

Comment:
How exactly do you need to deal with folders? ^^

Reply:
This is in regards to the implementation details of `fs::rename`.

> On Unix, if from is a directory, to must also be an (empty) directory. If from is not a directory, to must also be not a directory.

so, to be sane, it would be best to check what `run_northstar.txt` and its backup actually are.

Reply:
Ah so one could have a folder called `run_northstar.txt/` and then the function would actually fail.

Tbh, that's rather unlikely scenario and I'm more than happy to ignore it. If the function just panics in that case I could use Sentry dashboard to see if  / how often we hit it and apply a workaround if need be ^^

Reply:
Alright, I'll remove the comments when I find the time.